oak_functions/loader/benches/lookup.rs

```diff
@@ -33,125 +31,199 @@ use oak_functions_loader::{
+    let plot_config = PlotConfiguration::default().summary_scale(AxisScale::Logarithmic);
+    group.plot_config(plot_config);
+    for size in [16usize, 128usize, 1_024usize, 8_192usize].iter() {
+        let TestData {
+            test_key,
+            expected_value,
+            entries,
+        } = generate_random_test_data_for_bench(32, *size, 1_000);
+        let lookup_data = Arc::new(LookupData::for_test(entries.clone()));
+        let logger = Logger::for_test();
+        let wasm_handler = WasmHandler::create(&wasm_module_bytes, lookup_data, vec![], logger)
+            .expect("Couldn't create the server");
 
-#[bench]
-fn medium_random_data_single_lookup(bencher: &mut Bencher) {
-    let test_data = generate_random_test_data_for_bench(128, 256, 200_000);
+        let single_benchmark_request = BenchmarkRequest {
+            iterations: SINGLE_REQUEST_ITERATIONS,
+            action: Some(Action::Lookup(LookupTest {
+                key: test_key.clone(),
+            })),
+        }
+        .encode_to_vec();
+        let multi_benchmark_request = BenchmarkRequest {
+            iterations: MULTI_REQUEST_ITERATIONS,
+            action: Some(Action::Lookup(LookupTest {
+                key: test_key.clone(),
+            })),
+        }
+        .encode_to_vec();
 
-    run_lookup_bench(bencher, test_data, 1, Duration::from_millis(5));
+        group.bench_with_input(BenchmarkId::new("single", size), size, |b, _size| {
+            b.iter(|| {
+                run_lookup_iteration(
+                    &rt,
+                    &wasm_handler,
+                    &single_benchmark_request,
+                    &expected_value,
+                )
+            })
+        });
+        group.bench_with_input(BenchmarkId::new("multi", size), size, |b, _size| {
+            b.iter(|| {
+                run_lookup_iteration(
+                    &rt,
+                    &wasm_handler,
+                    &multi_benchmark_request,
+                    &expected_value,
+                )
+            })
+        });
+    }
+    group.finish();
 }
 
-#[bench]
-fn medium_random_data_multi_lookup(bencher: &mut Bencher) {
-    let test_data = generate_random_test_data_for_bench(128, 256, 200_000);
+fn entry_count(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+    let wasm_module_bytes = bulid_wasm_module(MANIFEST_PATH);
 
-    run_lookup_bench(bencher, test_data, 101, Duration::from_millis(10));
-}
+    let mut group = c.benchmark_group("entry_count");
+    let plot_config = PlotConfiguration::default().summary_scale(AxisScale::Logarithmic);
+    group.plot_config(plot_config);
+    for count in [100usize, 10_000usize, 1_000_000usize].iter() {
+        let TestData {
+            test_key,
+            expected_value,
+            entries,
+        } = generate_random_test_data_for_bench(64, 256, *count);
+        let lookup_data = Arc::new(LookupData::for_test(entries.clone()));
```

Comment:
It seems from here on the logic is the same in all the benches, perhaps extract to a function?

Reply:
Done.